etstool.py

```diff
@@ -268,6 +305,29 @@ def do_in_tempdir(files=(), capture_files=()):
         os.chdir(old_path)
         rmtree(path)
 
+@contextmanager
+def do_in_existingdir(path):
+    """ Changes into an existing directory given by path.
+    On exit, changes back to the original directory.
+
+    Parameters
+    ----------
+    path : str
+        Path of the directory to be changed into.
+    """
+    if not os.path.exists(path):
+        raise IOError("The path {!r} does not exist", path)
+    elif not os.path.isdir(path):
+        raise NotADirectoryError(
```

Comment:
`NotADirectoryError` isn't defined in Python 2; I'd like to keep the `etstool.py` script Python-version agnostic for now.

Actually, I'd suggest dropping these checks entirely; `os.chdir` already raises with a helpful error message in these cases, so there's little benefit in the extra checks.
```
>>> import os
>>> os.chdir("README.rst")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NotADirectoryError: [Errno 20] Not a directory: 'README.rst'
>>> os.chdir("bogus")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: 'bogus'
```